dbms/src/IO/WriteBufferFromS3.cpp

```diff
@@ -0,0 +1,286 @@
+#include <IO/WriteBufferFromS3.h>
+
+#include <IO/WriteHelpers.h>
+
+#include <Poco/DOM/AutoPtr.h>
+#include <Poco/DOM/DOMParser.h>
+#include <Poco/DOM/Document.h>
+#include <Poco/DOM/NodeList.h>
+#include <Poco/SAX/InputSource.h>
+
+#include <common/logger_useful.h>
+
+
+namespace DB
+{
+
+const int DEFAULT_S3_MAX_FOLLOW_PUT_REDIRECT = 2;
+const int S3_WARN_MAX_PARTS = 10000;
```

Comment:
A comment is needed...

Reply:
c45e7dc747b03a7091cb88da8eb3ebf83ad1fa13